lib/HTFeed/BackupExpiration.pm

```diff
@@ -59,16 +53,18 @@ SQL
         AND namespace=?
         AND id=?
         AND version < DATE_FORMAT(DATE_SUB(NOW(), INTERVAL 180 DAY),"%Y%m%d%H%i%S")
-        ORDER BY version DESC LIMIT 1,999999999
+        ORDER BY version DESC
 SQL
     $versions_sth->execute($self->{storage_name}, $row->{namespace}, $row->{id});
-    foreach my $version (map { $_->[0]; } @{$versions_sth->fetchall_arrayref}) {
+    my @versions = map { $_->[0]; } @{$versions_sth->fetchall_arrayref};
+    shift @versions; # jettison the most recent
```

Comment:
We might want to clarify the comment here -- "jettison" isn't perhaps the best word since we are in fact *retaining* that version.. perhaps:

```perl
shift @versions; # ensure we don't delete the most recent
```